sdk/python/kfp/components/executor_test.py

```diff
@@ -412,31 +413,40 @@ def test_func() -> NamedTuple('Outputs', [
                           ['output_dataset', 'output_int', 'output_string'])
       return output("Dataset contents", 101, "Some output string")
 
-    self._get_executor(test_func, executor_input).execute()
-    with open(os.path.join(self._test_dir, 'output_metadata.json'), 'r') as f:
-      output_metadata = json.loads(f.read())
-    self.assertDictEqual(
-        output_metadata, {
-            'artifacts': {
-                'output_dataset': {
-                    'artifacts': [{
-                        'metadata': {},
-                        'name': 'output_dataset',
-                        'uri': 'gs://some-bucket/output_dataset/data'
-                    }]
-                }
-            },
-            "parameters": {
-                "output_string": {
-                    "stringValue": "Some output string"
-                },
-                "output_int": {
-                    "intValue": 101
-                }
-            },
-        })
-
-    with open(os.path.join(self._test_dir, 'some-bucket/output_dataset/data'),
-              'r') as f:
-      artifact_payload = f.read()
-    self.assertEqual(artifact_payload, "Dataset contents")
+    # Functions returning plain tuples should work too.
+    def func_returning_plain_tuple() -> NamedTuple('Outputs', [
+        ("output_dataset", Dataset),
+        ("output_int", int),
+        ("output_string", str),
+    ]):
+      return ("Dataset contents", 101, "Some output string")
```

Comment:
Interesting. I would have thought this as a user code error for not returning an object of the declared return type (understand Python doesn't block this).
Feels to me what you did in this PR is more like a feature to add tolerance to the system than fixing a bug.

Reply:
Yeah exactly :-) In some sense, we need the named tuple so we can figure out the outputs and name them.

Reply:
I deliberated about this quite a lot.
On one hand, returning a simple tuple or a list is not exactly correct.

On the other hand, re-declaring the NamedTuple is too much boilerplate.

Simple tuples are used a lot as Python return values, so I've decided to enable this style as well (the declaration still needs NamedTuple to give names and types to the outputs).